src/main/kotlin/no/nav/familie/ef/sak/service/PerioderOvergangsstønadService.kt

```diff
@@ -2,31 +2,64 @@ package no.nav.familie.ef.sak.service
 
 import no.nav.familie.ef.sak.integration.InfotrygdReplikaClient
 import no.nav.familie.ef.sak.integration.PdlClient
+import no.nav.familie.ef.sak.repository.domain.SøknadType
+import no.nav.familie.kontrakter.ef.felles.StønadType
 import no.nav.familie.kontrakter.ef.infotrygd.InfotrygdPerioderOvergangsstønadRequest
+import no.nav.familie.kontrakter.ef.infotrygd.StønadTreff
+import no.nav.familie.kontrakter.ef.infotrygd.SøkFlereStønaderRequest
 import no.nav.familie.kontrakter.felles.ef.PeriodeOvergangsstønad
 import no.nav.familie.kontrakter.felles.ef.PerioderOvergangsstønadRequest
 import no.nav.familie.kontrakter.felles.ef.PerioderOvergangsstønadResponse
+import no.nav.familie.kontrakter.felles.objectMapper
 import org.springframework.stereotype.Service
 
 @Service
 class PerioderOvergangsstønadService(private val infotrygdReplikaClient: InfotrygdReplikaClient,
                                      private val pdlClient: PdlClient) {
 
+    /**
+     * Henter perioder fra infotrygd for en person
+     */
     fun hentPerioder(request: PerioderOvergangsstønadRequest): PerioderOvergangsstønadResponse {
-        val personIdenter = pdlClient.hentPersonidenter(request.personIdent, true).identer.map { it.ident }.toSet()
-        val perioder = hentPerioderFraInfotrygd(personIdenter, request)
-        return PerioderOvergangsstønadResponse(perioder)
-    }
-
-    private fun hentPerioderFraInfotrygd(personIdenter: Set<String>,
-                                         request: PerioderOvergangsstønadRequest): List<PeriodeOvergangsstønad> {
+        val personIdenter = hentIdenter(request.personIdent)
         val infotrygdRequest = InfotrygdPerioderOvergangsstønadRequest(personIdenter, request.fomDato, request.tomDato)
         val infotrygdPerioder = infotrygdReplikaClient.hentPerioderOvergangsstønad(infotrygdRequest)
-        return infotrygdPerioder.perioder.map {
+        val perioder = infotrygdPerioder.perioder.map {
             PeriodeOvergangsstønad(personIdent = it.personIdent,
                                    fomDato = it.fomDato,
                                    tomDato = it.tomDato,
                                    datakilde = PeriodeOvergangsstønad.Datakilde.INFOTRYGD)
         }
+        return PerioderOvergangsstønadResponse(perioder)
+    }
+
+    fun eksisterer(ident: String, søknadType: SøknadType): StønadTreff {
+        val response = eksisterer(ident, setOf(søknadType))
+        val stønadTreff = response[søknadType]
+        requireNotNull(stønadTreff) {
+            "Forventer att man skal få treff på $søknadType men fikk ${
+                objectMapper.writeValueAsString(response)
+            }"
+        }
+        return stønadTreff
     }
+
+    /**
+     * Finner om en person eksisterer i infotrygd og om det finnes noen aktiv sak for personen.
+     */
+    fun eksisterer(ident: String, søknadTyper: Set<SøknadType>): Map<SøknadType, StønadTreff> {
```

Comment:
Klassen heter `PerioderOvergangsstønadService`, men vi kan sjekke om en person eksisterer for alle typer stønader. Burde klassen bytte navn og funksjonen `hentPerioder` heller hete `hentPerioderForOvergangsstønad` ? Evt. trekke ut eksisterer til stønadsuavhengig klasse?

Reply:
Splitte i to servicer? 
Og sjekker vi om det eksisterer en person, om det eksisterer en fagsak, eller om det eksisterer en løpende behandling? (navngivning). Kanskje finnesAktivSakForPerson?

Reply:
Ser vi for oss at denne servicen skal merge våre statuser med infotrygds etterhvert, eller vi det skje utenfor? Hvis dette bare er for å hondtere infotrygd kunne vi kanskje tatt med dette når vi setter navn?

Reply:
Ja her hadde det gått helt galt, flyttet ut den til egen service. PerioderService skal hente perioder fra ef-sak senere også